core/src/main/scala/org/bitcoins/core/crypto/SchnorrDigitalSignature.scala

```diff
@@ -0,0 +1,71 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.{BitcoinSLogger, BitcoinSUtil, Factory}
+import scodec.bits.ByteVector
+
+sealed abstract class SchnorrDigitalSignature extends NetworkElement {
```

Comment:
This could just be a direct case class IMO, no need for the abstract class/companion object/Impl pattern (leads to more allocation as well as more verbose code)

Reply:
If this were a case class then I couldn't have the `DummySchnorrDigitalSignature` could I?

Reply:
As long as the object is not a `case object` it works out nicely: https://github.com/bitcoin-s/bitcoin-s/blob/9101aece9b9e48df741e668fce66edbf02f59afc/core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionOutPoint.scala#L32

Reply:
We probably want to at least make this a trait, so we can `extend AnyVal` in `SchnorrDigitalSignatureImpl()` later on to avoid runtime allocations.